kafka: add documentation + more configurations (#41)
kudobuilder/frameworks: repository/kafka/docs/v0.1/monitoring.md

# Monitoring

> KUDO Kafka operator comes by default the JMX Exporter agent enabled. 

Reviewer: ```suggestion
KUDO Kafka operator comes by default with the JMX Exporter agent enabled. 
```

> When Kafka operator deployed with parameter `METRICS_ENABLED=true` (which defaults to `true`) then:

Reviewer: ```suggestion
When the Kafka operator is deployed with the parameter `METRICS_ENABLED=true` (which defaults to `true`) then:
```

> - Each broker bootstraps with [JMX Exporter](https://github.com/prometheus/jmx_exporter) java agent exposing the metrics at `9094/metrics`

Reviewer: ```suggestion
- Each broker bootstraps with [JMX Exporter](https://github.com/prometheus/jmx_exporter) Java agent exposing the metrics at `9094/metrics`.
```

> - Adds a port named `metrics` to the Kafka Service

Reviewer: ```suggestion
- Adds a port named `metrics` to the Kafka Service.
```

- Adds a label `kubeaddons.mesosphere.io/servicemonitor: "true"` for the service monitor discovery. 

```
kubectl describe svc kafka-svc
...
Port:              metrics  9094/TCP
TargetPort:        9094/TCP
...
```

### Using Prometheus Service Monitor

> To use the prometheus service monitor, its necessary to have installed the prometheus operator previously in the cluster.

Reviewer: ```suggestion
To use the Prometheus Service Monitor, it is necessary to have the Prometheus operator previously installed in the cluster.
```

Which Prometheus Operator are we referring to here? Can we link to the docs?

> If the Kafka Cluster Service in the default namespace, we can use the next example of the service-monitor.

Reviewer: ```suggestion
If the Kafka Cluster Service is in the default namespace, we can use the next example of the service-monitor.
```

```
kubectl create -f resources/service-monitor.yaml
```

> Install the [grafana dashboard](./resources/grafana-dashboard.json), and we should be able to see the Kafka dashaboard.

Reviewer: ```suggestion
Install the [grafana dashboard](./resources/grafana-dashboard.json), and we should be able to see the Kafka dashboard.
```
